### ISSUE TITLE
proposal: change the default value of nsqd.persistence.storageClass to ""

### ISSUE THREAD

**Issue by wjiec** [nsqio/helm-chart#21]
First of all, thank you for this exquisite program!

> **The following is based on the premise that most kubernetes clusters have a default StorageClass, which I think should be reasonable.**

Currently, the default value for `nsqd.persistence.storageClass` is `“-”`, which causes the pvc to remain in the pending state, requiring the pv to be manually created in order for nsq to continue to run.

To improve the usability and flexibility of the charts, I recommend changing the default value of `nsqd.persistence.storageClass` to the empty string (“”). This way, Helm charts will automatically utilize the cluster's default StorageClass, simplifying deployment and reducing the configuration burden on the user.

**Comment by ploxiln** [comment]
{{- if (ne "-" .Values.nsqd.persistence.storageClass) }}
      storageClassName: "{{ .Values.nsqd.persistence.storageClass }}"
    {{- end }}

It seems that "-" is meant to just omit this specification ... perhaps that was meant to use the default storageClass? Did this behavior change in later k8s versions, or does it vary by provider (EKS vs GKE etc)? Does it makes sense to get rid of that conditional and just make the default value `""`, or is there still some case where someone may need to completely omit that line?

**Comment by ploxiln** [comment]
It's a bit confusing, but it seems like this chart already does the right thing to use the defaultStorageClass:

> If or when a default StorageClass becomes available, the control plane identifies any existing PVCs without storageClassName. For the PVCs that either have an empty value for storageClassName or do not have this key, the control plane then updates those PVCs to set storageClassName to match the new default StorageClass. If you have an existing PVC where the storageClassName is "", and you configure a default StorageClass, then this PVC will not get updated.

https://kubernetes.io/docs/concepts/storage/storage-classes/#default-storageclass

**Comment by wjiec** [comment]
Awwww, sorry, I was looking at the wrong repository 😅 I had too many tabs open when I was doing the deployment and I was looking at the wrong content: https://github.com/BeeInventor/charts/blob/7988e3e3722bbbd67ac4ff0b1a4492615186f1ac/beeinventor/nsq/templates/nsqd-statefulset.yaml#L143-L149

Thanks for your patience, you are right! By the way, I think we could take a page from bitnami and make it better, do you think?
https://github.com/bitnami/charts/blob/f08566c38dfd988cc6e87c4e925e43d21f381b95/bitnami/common/templates/_storage.tpl#L8-L21

TL;DR: “-” means use manually created pv, “” means use default StorageClass, and others means specify specific StorageClass.

**Comment by ploxiln** [comment]
Hmm ... I don't think it's worth adding more logic to. I guess it happens to be a bit opposite of the bitnami pattern, here: "-" for default, "" for manually-created pv? I think it's more direct here :shrug:
